config/v1/types_infrastructure.go

```diff
@@ -229,6 +229,17 @@ const (
 	IBMCloudProviderTypeUPI IBMCloudProviderType = "UPI"
 )
 
+// ClusterHostedDNS indicates whether the cluster DNS is hosted by the cluster or Core DNS .
+type ClusterHostedDNS string
+
+const (
+	// ClusterHostedDNSEnabled indicates that the DNS solution is hosted by the cloud platform.
+	ClusterHostedDNSEnabled ClusterHostedDNS = "Enabled"
+
+	// ClusterHostedDNSDisabled indicates that the cluster is using the DNS solution provided by the cloud platform.
+	ClusterHostedDNSDisabled ClusterHostedDNS = "Disabled"
```

Comment:
Reading the description of these two, I concluded they do the same thing... I think these need to be updated/made more clear in their intention

Reply:
`const (
	
	// ClusterHostedDNSDisabled indicates that the cluster is using the default DNS solution for the cloud platform.
	ClusterHostedDNSDisabled ClusterHostedDNS = "Disabled"`

        // ClusterHostedDNSEnabled indicates that a DNS solution other than the default provided by the cloud platform is in use. In this mode, the cluster hosts a DNS solution during installation and the user is expected to provide their own DNS solution post-install. Both the cluster's self-hosted DNS and the user's DNS solution are not expected to be the one provided by the underlying cloud.
	ClusterHostedDNSEnabled ClusterHostedDNS = "Enabled"

)`
@barbacbd how about this? IMHO, defining the default "Disabled" sets the stage to better understand what `Enabled` stands for.

Reply:
+1 Seems good to me. @JoelSpeed what do you think?

Reply:
Can we expand the first comment for disabled to explain that this means that, in installer-provisioned-infrastructure clusters on cloud providers this typically means the installer was responsible for creating and configuring the DNS?

Is the distinction here that we are trying to make really that I wouldn't be using route53 on AWS? I'm sure I could probably configure route53 and use this clusterHostedDNS feature right? Is there a use case for that?

Reply:
> Can we expand the first comment for disabled to explain that this means that, in installer-provisioned-infrastructure clusters on cloud providers this typically means the installer was responsible for creating and configuring the DNS?
> 
@JoelSpeed I want to specify this mode in terms of OpenShift behavior as a whole and not what a specific component does. So, yes, OpenShift configures the underlying cloud provided DNS. For example Route53 in AWS. 

> Is the distinction here that we are trying to make really that I wouldn't be using route53 on AWS? I'm sure I could probably configure route53 and use this clusterHostedDNS feature right? Is there a use case for that?

Yes, the target use case is a set of customers who cannot use the cloud provided DNS due to ITAR requirements. It is in the enhancement. If the customer enables this feature, Installer will not configure Route53. The Installer now only configures the LB and provides the MCO with the LB IPs so that it can start the in-cluster DNS. (Again, I don't want to specify what in-cluster DNS solution is being used in the API.) If the customer then configures Route53 on their own post-install, we don't really care. Nothing in the code will prevent such a usage but we cannot at this time predict every scenario where this feature can be used.

Reply:
// ClusterHostedDNSDisabled indicates that the cluster is using the default DNS solution for the
       // cloud platform. OpenShift is responsible for all the LB and DNS configuration needed for the
       // cluster to be functional with no intervention from the user. To accomplish this, OpenShift
       // configures the default LB and DNS solutions provided by the underlying cloud.
       
       @JoelSpeed, @barbacbd does this comment explain things better?

Reply:
ClusterHostedDNSDisabled - Uses default cloud provided DNS (Example: Route53 for DNS). OpenShif does all the configuration.

ClusterHostedDNSEnabled - OpenShift does **not** use default cloud provided DNS. Starts an in-cluster DNS solution for installation success. User/Customer can augment with a custom DNS solution post-install in scenarios where regulations require them to.

Reply:
> // ClusterHostedDNSDisabled indicates that the cluster is using the default DNS solution for the
// cloud platform. OpenShift is responsible for all the LB and DNS configuration needed for the
// cluster to be functional with no intervention from the user. To accomplish this, OpenShift
// configures the default LB and DNS solutions provided by the underlying cloud.

I like this explanation, can we include?

I'm not sure I like the comment in the Enabled case that says it's expected not to be the cloud provider default. That seems like an odd comment to me that is likely to raise questions

Reply:
Comment for Enabled case has been updated to remove that.

Even in the original comment, we wanted to highlight that **OpenShift will** bypass configuring the default cloud provided DNS. That is central to the requirements of this feature. What the customer does post-install should not be part of this comment.